core/src/main/java/io/questdb/griffin/engine/functions/catalogue/CurrentUserFunctionFactory.java

```diff
@@ -45,6 +53,31 @@ public boolean isRuntimeConstant() {
 
     @Override
     public Function newInstance(int position, ObjList<Function> args, IntList argPositions, CairoConfiguration configuration, SqlExecutionContext sqlExecutionContext) {
-        return Constants.USER_CONSTANT;
+        return new CurrentUserFunction();
+    }
+
+    static class CurrentUserFunction extends StrFunction {
+        private SecurityContext context;
+
+        @Override
+        public CharSequence getStr(Record rec) {
+            return context.getPrincipal();
```

Comment:
noticed that `getEntityName()` has been renamed to `getPrincipal()`.
this will be either confusing or incorrect, depending on how the method will be implemented.
if we cannot keep `getEntityName()`, let's call it `getCurrentUser()` or `getCurrentEntity()`?

Reply:
sorry, this is me being a bear in a china shop, sorry!

Reply:
I changed it back to entityName.